iceoryx_binding_c/include/iceoryx_binding_c/internal/cpp2c_subscriber.hpp

```diff
@@ -15,15 +15,33 @@
 #ifndef IOX_BINDING_C_CPP2C_SUBSCRIBER_H
 #define IOX_BINDING_C_CPP2C_SUBSCRIBER_H
 
+#include "iceoryx_binding_c/enums.h"
 #include "iceoryx_posh/internal/popo/ports/subscriber_port_user.hpp"
-#include "iceoryx_posh/popo/condition.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
 
-struct cpp2c_Subscriber : public iox::popo::Condition
+struct cpp2c_Subscriber
 {
-    void setConditionVariable(iox::popo::ConditionVariableData* const conditionVariableDataPtr) noexcept override;
-    bool hasTriggered() const noexcept override;
-    void unsetConditionVariable() noexcept override;
+    cpp2c_Subscriber() = default;
+    cpp2c_Subscriber(const cpp2c_Subscriber&) = delete;
+    cpp2c_Subscriber(cpp2c_Subscriber&& rhs) = delete;
+    ~cpp2c_Subscriber();
+
+    cpp2c_Subscriber& operator=(const cpp2c_Subscriber&) = delete;
+    cpp2c_Subscriber& operator=(cpp2c_Subscriber&& rhs) = delete;
+
+    iox_WaitSetResult attachToWaitset(iox::popo::WaitSet& waitset,
+                                      const iox_SubscriberEvent event,
+                                      const uint64_t triggerId,
+                                      const iox::popo::Trigger::Callback<cpp2c_Subscriber> callback = nullptr) noexcept;
+
+    void detachWaitset() noexcept;
+
+    void unsetTrigger(const iox::popo::Trigger& trigger) noexcept;
+
+    bool hasNewSamples() const noexcept;
+
 
     iox::popo::SubscriberPortData* m_portData{nullptr};
+    iox::popo::Trigger m_trigger;
```

Comment:
This would be one option. But I would prefer in this case one or two more member variables since we only have 2 or 3 event types per class and it would be much more readable. In the end it could look like:
```cpp
class Subscriber {
   Trigger m_hasNewSamplesTrigger;
   Trigger m_hasLostSamplesTrigger;
   Trigger m_isSubscribedTrigger;
};
```

If we would use a `cxx::vector` we would have much more developer overhead since we have to track which index corresponds with which trigger etc.